src/Markdown.MAML/Renderer/MamlRenderer.cs

```diff
@@ -75,331 +31,161 @@ public string MamlModelToString(IEnumerable<MamlCommand> mamlCommands)
         {
-            PushTag("command:returnValues");
-            foreach (var output in command.Outputs)
-            {
-                PushTag("command:returnValue");
-
-                PushTag("dev:type");
-                PushTag("maml:name");
-                _stringBuilder.Append(XmlEscape(output.TypeName));
-                PopTag("maml:name");
-                PopTag("dev:type");
-
-                PushTag("maml:description");
-                AddParas(output.Description);
-                PopTag("maml:description");
-
-                PopTag("command:returnValue");
-            }
-            PopTag("command:returnValues");
+            return new XElement(commandNS + "inputType",
+                    new XElement(devNS + "type",
+                        new XElement(mamlNS + "name", input.TypeName)),
+                    new XElement(mamlNS + "description", GenerateParagraphs(input.Description)));
         }
 
-        private void AddInputs(MamlCommand command)
+        private static XElement CreateOutput(MamlInputOutput output)
         {
-            PushTag("command:inputTypes");
-            foreach (var input in command.Inputs)
-            {
-                PushTag("command:inputType");
-
-                PushTag("dev:type");
-                PushTag("maml:name");
-                _stringBuilder.Append(XmlEscape(input.TypeName));
-                PopTag("maml:name");
-                PopTag("dev:type");
-
-                PushTag("maml:description");
-                AddParas(input.Description);
-                PopTag("maml:description");
-
-                PopTag("command:inputType");
-            }
-            PopTag("command:inputTypes");
+            return new XElement(commandNS + "returnValue",
+                    new XElement(devNS + "type",
+                        new XElement(mamlNS + "name", output.TypeName)),
+                    new XElement(mamlNS + "description", GenerateParagraphs(output.Description)));
         }
 
-        private void AddParameters(MamlCommand command)
+        private static XElement CreateExample(MamlExample example)
         {
-            PushTag("command:parameters");
-            foreach (MamlParameter parameter in command.Parameters)
-            {
-                AddParameter(command, parameter, inSyntax:false);
-            }
-            PopTag("command:parameters");
+            return new XElement(commandNS + "example",
+                    new XElement(mamlNS + "title", example.Title),
+                    new XElement(devNS + "code", example.Code),
+                    new XElement(devNS + "remarks", GenerateParagraphs(example.Remarks)));
         }
 
-        private string ConvertPSTypeToMamlType(MamlParameter parameter)
+        private static XElement CreateLink(MamlLink link)
         {
-            if (parameter.Type == null)
+            if(link.IsSimplifiedTextLink)
             {
-                return "";
+                return null;
```

Comment:
Yes. Any nulls are just passed over and not rendered as anything. Same applies to empty lists